funds-manager/funds-manager-server/src/fee_indexer/fee_balances.rs

```diff
@@ -38,8 +87,88 @@ impl Indexer {
             derive_wallet_keychain(&eth_key, self.chain_id).map_err(FundsManagerError::custom)?;
         let root_key = wallet_keychain.secret_keys.sk_root.clone().expect("root key not present");
 
-        // Fetch the wallet from the relayer
-        let wallet = self.relayer_client.get_wallet(wallet_metadata.id, &root_key).await?;
-        Ok(wallet.wallet)
+        // Fetch the wallet from the relayer and replace the keychain so that we have
+        // access to the full set of secret keys
+        let mut wallet =
+            self.relayer_client.get_wallet(wallet_metadata.id, &root_key).await?.wallet;
+        let keychain =
+            derive_wallet_keychain(&eth_key, self.chain_id).map_err(FundsManagerError::custom)?;
```

Comment:
do we need to re-derive the keychain here? we can just use the one derived on line 86